server/server_test.go

```diff
@@ -70,55 +71,82 @@ func TestServer_WithBaseContext(t *testing.T) {
 	}
 }
 
-func TestServer_Run(t *testing.T) {
-	// Create a new Server instance
-	server := NewServer(":0")
+func TestServer_WithReadinessChan(t *testing.T) {
+	// Create a new Server instance with a base context
+	ready := make(chan struct{})
+	server := NewServer(":0", WithReadinessChan(ready))
 
-	channel := mocks.NewMockChannel(t)
-	channel.EXPECT().Path().Return("/test")
-	channel.EXPECT().Handler().Return(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
+	if server.ready == nil {
+		t.Error("Expected non-nil channel")
+	}
 
-	server.AddChannel(channel)
+	close(server.ready)
+	_, ok := <-ready
+	if ok {
+		t.Error("Expected closed channel")
+	}
+}
 
-	done := make(chan struct{})
+func TestServer_Run(t *testing.T) {
 
-	// Run the server
-	for i := 0; i < 2; i++ {
-		go func() {
-			err := server.Run()
-			switch err {
-			case nil:
-				close(done)
-			case ErrServerAlreadyRunning:
-				done <- struct{}{}
-			default:
-				t.Errorf("Got unexpected error: %v", err)
+	noOfReruns := []int{0, 1, 2, 5}
```

Comment:
not sure it gives additional value

Reply:
calling Run initially should not throw error, but repeatedly should throws error, this check test for that scenario. 5 might be unnecessary though so I removed it